apps/studio/components/ui/AIAssistantPanel/AIAssistantPanel.tsx

```diff
@@ -0,0 +1,439 @@
+import { useQueryClient } from '@tanstack/react-query'
+import { uniqBy } from 'lodash'
+import { Command, CornerDownLeft, Loader2, X } from 'lucide-react'
+import { useRouter } from 'next/router'
+import { useCallback, useEffect, useRef, useState } from 'react'
+import { toast } from 'sonner'
+
+import { useChat } from 'ai/react'
+import { useParams } from 'common'
+import { generateThreadMessage } from 'components/interfaces/Auth/Policies/AIPolicyEditorPanel/AIPolicyEditorPanel.utils'
+import { MessageWithDebug } from 'components/interfaces/SQLEditor/AiAssistantPanel'
+import { sqlAiDisclaimerComment } from 'components/interfaces/SQLEditor/SQLEditor.constants'
+import { DiffType, IStandaloneCodeEditor } from 'components/interfaces/SQLEditor/SQLEditor.types'
+import { suffixWithLimit } from 'components/interfaces/SQLEditor/SQLEditor.utils'
+import Results from 'components/interfaces/SQLEditor/UtilityPanel/Results'
+import { useSqlDebugMutation } from 'data/ai/sql-debug-mutation'
+import { databasePoliciesKeys } from 'data/database-policies/keys'
+import { QueryResponseError, useExecuteSqlMutation } from 'data/sql/execute-sql-mutation'
+import { sqlKeys } from 'data/sql/keys'
+import { usePrevious } from 'hooks/deprecated'
+import { useLocalStorage } from 'hooks/misc/useLocalStorage'
+import { useSelectedProject } from 'hooks/misc/useSelectedProject'
+import { SqlEditor } from 'icons'
+import { BASE_PATH, LOCAL_STORAGE_KEYS } from 'lib/constants'
+import { detectOS, uuidv4 } from 'lib/helpers'
+import { useAppStateSnapshot } from 'state/app-state'
+import {
+  Button,
+  cn,
+  Sheet,
+  SheetContent,
+  SheetFooter,
+  SheetHeader,
+  Tooltip_Shadcn_,
+  TooltipContent_Shadcn_,
+  TooltipTrigger_Shadcn_,
+} from 'ui'
+import { Admonition } from 'ui-patterns'
+import CodeEditor from '../CodeEditor/CodeEditor'
+import { AIAssistant } from './AIAssistant'
+import { generateCTA, generatePlaceholder, generateTitle, validateQuery } from './AIAssistant.utils'
+import { ASSISTANT_SUPPORT_ENTITIES } from './AiAssistant.constants'
+
+export const AiAssistantPanel = () => {
+  const os = detectOS()
+  const router = useRouter()
+  const { ref } = useParams()
+  const project = useSelectedProject()
+  const queryClient = useQueryClient()
+  const { aiAssistantPanel, setAiAssistantPanel } = useAppStateSnapshot()
+
+  const { open, editor, content } = aiAssistantPanel
+  const previousEditor = usePrevious(editor)
+
+  const [isAcknowledged, setIsAcknowledged] = useLocalStorage(
+    LOCAL_STORAGE_KEYS.SQL_SCRATCH_PAD_BANNER_ACKNOWLEDGED,
+    false
+  )
+
+  const [chatId, setChatId] = useState(uuidv4())
```

Comment:
Here to micro-optimise all the things 😅
```suggestion
  const [chatId, setChatId] = useState(() => uuidv4())
```